File: EntityLayer/Entities/AssetZoneHistory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EntityLayer.Entities
{
    public class AssetZoneHistory
    {
        [Key]
        public int Id { get; set; }
        public int AssetId { get; set; }
        public int ZoneId { get; set; }
        public DateTime EnterDateTime { get; set; }
        public DateTime? ExitDateTime { get; set; }
        public TimeSpan? RetentionTime { get; set; }

        public virtual Asset Asset { get; set; } = null!;
        public virtual Zone Zone { get; set; } = null!;
    }
}

File: Android/Software/core/src/main/java/hr/foi/air/core/models/impl/AssetZoneHistory.kt
package hr.foi.air.core.models.impl

import hr.foi.air.core.models.IAssetZoneHistory

class AssetZoneHistory(
    override var id: Int,
    override val assetId: Int,
    override val zoneId: Int,
    override val enterDateTime: String,
    override val exitDateTime: String?,
    override val retentionTime: String
):IAssetZoneHistory {
}